### PR TITLE
Adds a unit tests for scale-up annotation.

### PR DESCRIPTION
**Pull request by ishan16696**
/area testing
/kind test

**What this PR does / why we need it**:
This PR adds a unit tests for scale-up various scenario:
1. when scale-up is configured a scale-up annotation: `gardener.cloud/scaled-to-multi-node` should be added to etcd statefulset.
2. when scale-up is configured but scale-up isn't completed/successful, a scale-up annotation: `gardener.cloud/scaled-to-multi-node` shouldn't be removed from etcd statefulset.
3. when scale-up is successful then a scale-up annotation: `gardener.cloud/scaled-to-multi-node` should be removed from the etcd statefulset.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:
```other operator
None
```

### CODE REVIEW

File: pkg/component/etcd/statefulset/statefulset_test.go [hunk 1]
```diff
@@ -175,52 +174,45 @@ var _ = Describe("Statefulset", func() {
 
 	Describe("#Deploy", func() {
 		Context("when statefulset does not exist", func() {
-			It("should create the statefulset successfully", func() {
-				Expect(stsDeployer.Deploy(ctx)).To(Succeed())
+			Context("bootstrap a single replica statefulset", func() {
```
Comment (shreyas-s-rao) [inline pkg/component/etcd/statefulset/statefulset_test.go:177]:
```suggestion
			Context("bootstrap of a single replica statefulset", func() {
```

File: pkg/component/etcd/statefulset/statefulset_test.go [hunk 2]
```diff
@@ -175,52 +174,45 @@ var _ = Describe("Statefulset", func() {
...
 
-				Expect(cl.Get(ctx, kutil.Key(namespace, values.Name), sts)).To(Succeed())
-				checkStatefulset(sts, values)
+					Expect(cl.Get(ctx, kutil.Key(namespace, values.Name), sts)).To(Succeed())
+					checkStatefulset(sts, values)
+				})
 			})
 
-			Context("should bootstrap a multi replica statefulset successfully", func() {
+			Context("bootstrap a multi replica statefulset successfully", func() {
```
Comment (shreyas-s-rao) [inline pkg/component/etcd/statefulset/statefulset_test.go:188]:
```suggestion
			Context("bootstrap of a multi replica statefulset", func() {
```

File: pkg/component/etcd/statefulset/statefulset_test.go [hunk 3]
```diff
@@ -233,17 +225,134 @@ var _ = Describe("Statefulset", func() {
...
 
-				Expect(cl.Get(ctx, kutil.Key(namespace, values.Name), sts)).To(Succeed())
-				checkStatefulset(sts, values)
+				Expect(cl.Get(ctx, kutil.Key(namespace, values.Name), updatedSts)).To(Succeed())
+				checkStatefulset(updatedSts, values)
 			})
 
 			Context("when multi-node cluster is configured", func() {
+				var (
+					annotation = make(map[string]string)
```
Comment (shreyas-s-rao) [inline pkg/component/etcd/statefulset/statefulset_test.go:236]:
```suggestion
					annotations = make(map[string]string)
```

**Review by shreyas-s-rao — APPROVED** [review]
@ishan16696 thanks for adding these tests. Other than few cosmetic nits, LGTM

/lgtm

**Review by aaronfern — APPROVED** [review]
Thanks for the tests @ishan16696!
/lgtm

**Pull request merged**